### PR TITLE
Feature/#570 pin code protection

### PR DESCRIPTION
**Pull request by antonm76**
Simulator Screen Shot - iPhone SE - 2019-04-15 at 15 00 15

### CODE REVIEW

File: mobile/android/app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -4,6 +4,7 @@
     <uses-permission android:name="android.permission.INTERNET" />
     <uses-permission android:name="android.permission.SYSTEM_ALERT_WINDOW"/>
     <uses-permission android:name="android.permission.USE_FINGERPRINT"/>
+    <uses-permission android:name="android.permission.VIBRATE"/>
```
Comment (YaroTs) [inline mobile/android/app/src/main/AndroidManifest.xml]: Why is it needed?

File: mobile/android/app/src/main/java/ru/spb/arcadia/product/assistant/android/MainApplication.java [hunk 2]
```diff
@@ -42,6 +43,7 @@ public boolean getUseDeveloperSupport() {
         protected List<ReactPackage> getPackages() {
             return Arrays.asList(
                     new MainReactPackage(),
+            new AsyncStoragePackage(),
```
Comment (YaroTs) [inline mobile/android/app/src/main/java/ru/spb/arcadia/product/assistant/android/MainApplication.java]: Bad formatting

File: mobile/src/auth/pin-code-storage.ts [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+//============================================================================
+export class PinCodeStorage {
+    private static options = {
+        keychainService: Storage.keychain,
+        sharedPreferencesName: Storage.preferences,
+    };
+
+    private pin: string | null = null;
+
```
Comment (YaroTs) [inline mobile/src/auth/pin-code-storage.ts]: Why don't you use Nullable which is used in the rest of the project?

File: mobile/src/auth/pin-code-storage.ts [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    //----------------------------------------------------------------------------
+    public async getPin(): Promise<string | null> {
+        if (this.pin !== null) {
+            return this.pin;
```
Comment (YaroTs) [inline mobile/src/auth/pin-code-storage.ts]: Just if(this.pin) seems to be enough

File: mobile/src/auth/pin-code-storage.ts (hunk 3 above)
Comment (DesmanLead) [reply mobile/src/auth/pin-code-storage.ts]: @YaroTs stage 5, acceptance.

**Comment by antonm76** [comment]
Screenshot 2019-04-15 at 20 16 20
Simulator Screen Shot - iPhone 5s - 2019-04-15 at 20 17 07
Simulator Screen Shot - iPhone 5s - 2019-04-15 at 20 17 15
Simulator Screen Shot - iPhone 5s - 2019-04-15 at 20 17 43
Simulator Screen Shot - iPhone 5s - 2019-04-15 at 20 18 12

**Comment by antonm76** [comment]
Screenshot 2019-04-15 at 20 16 20
Simulator Screen Shot - iPhone 5s - 2019-04-15 at 20 17 07
Simulator Screen Shot - iPhone 5s - 2019-04-15 at 20 17 15
Simulator Screen Shot - iPhone 5s - 2019-04-15 at 20 17 43
Simulator Screen Shot - iPhone 5s - 2019-04-15 at 20 18 12

**Comment by antonm76** [comment]
Simulator Screen Shot - iPhone 5s - 2019-04-15 at 20 17 07

**Comment by antonm76** [comment]
Simulator Screen Shot - iPhone 5s - 2019-04-15 at 20 17 15

**Comment by antonm76** [comment]
Simulator Screen Shot - iPhone 5s - 2019-04-15 at 20 17 43

**Comment by antonm76** [comment]
Simulator Screen Shot - iPhone 5s - 2019-04-15 at 20 18 12

**Pull request merged**